### DISCUSSION TITLE
unable to load Dicom.Native-win-x64.dll

### DISCUSSION THREAD

**Discussion by erlaxmikantha** [fo-dicom/fo-dicom#1747] [Q&A] [answered]
It is observed that application using fo-dicom, fo-dicom.Codecs and fo-dicom.Imaging.ImageSharp often throws exception that "**unable to load Dicom.Native-win-x64.dll**" though this dll is present in application folder. 
I am using following nugets
```
    <PackageVersion Include="fo-dicom" Version="5.0.3" />
    <PackageVersion Include="fo-dicom.Codecs" Version="5.10.3" />
    <PackageVersion Include="fo-dicom.Imaging.ImageSharp" Version="5.0.3" />
```
This error vanished after installing vc++ redistributable latest version.
dumpbin /dependents indicated that Dicom.Native-win-x64.dll having dependency on msvcp140.dll and vcruntime140.dll from vc++ redistributable.
**Is it not sufficient to copy only msvcp140.dll and vcruntime140.dll into application directory instead of installing whole vc++ redistributable?**

**Comment by mrbean-bremen** [comment, marked as answer]
It _might_ be sufficient, but I would not recommend it. The whole VC++ redistributable is not large, it's only a few DLLs, so it should not be a problem spacewise. 

Copying the DLLs manually into the application directory has a few disadvantages:
- you won't get updated versions, for example with security fixes
- the DLLs would not be protected (e.g. automatically copied back if damaged)
- for the next application that needs them, you would have to duplicate them in that application directory again
- the next version of fo-dicom (or even the current one) may need more of the DLLs that are not directly shown as dependent

There are probably more reasons, but you get the gist...

**Reply by erlaxmikantha** [reply]
Is there any link to the document which states this library needs vc++ redistributable? I could see one readme of fo-dicom.Codecs
https://github.com/Efferent-Health/fo-dicom.Codecs/blob/master/README.md which states that "The codecs in this repository are written in pure C/C++ code and wrapped with C# and netstandard2.0"

**Reply by mrbean-bremen** [reply]
I think not, but this is probably something that would make sense to have in `fo-dicom.Codecs`, and linked to from (or copied to) `fo-dicom`.
